server/src/main/java/org/elasticsearch/index/mapper/GeoPointFieldMapper.java

```diff
@@ -110,17 +137,28 @@ public FieldMapper build(ContentPath contentPath) {
                 (ParsedGeoPoint) nullValue.get(),
                 ignoreZValue.get().value(),
                 ignoreMalformed.get().value());
-            GeoPointFieldType ft
-                = new GeoPointFieldType(buildFullName(contentPath), indexed.get(), stored.get(), hasDocValues.get(), geoParser, meta.get());
-            return new GeoPointFieldMapper(name, ft, multiFieldsBuilder.build(this, contentPath),
-                copyTo.build(), geoParser, this);
+            GeoPointFieldType ft = new GeoPointFieldType(
+                buildFullName(contentPath),
+                indexed.get(),
+                stored.get(),
+                hasDocValues.get(),
+                geoParser,
+                scriptValues(),
+                meta.get());
+            if (this.script.get() == null) {
+                return new GeoPointFieldMapper(name, ft, multiFieldsBuilder.build(this, contentPath),
+                    copyTo.build(), geoParser, this);
+            }
+            return new GeoPointFieldMapper(name, ft, geoParser, this);
```

Comment:
is it necessary to use two different constructors with or without the script?

Reply:
The class hierarchy of the geo mappers makes things complicated here, but I can change this so that there's just a single constructor at the expense of adding extra params to all the other class constructors if you think it's less confusing.

Reply:
I don't have a strong opinion and I cannot judge if it makes things better. Your call ;)